ci/common/build.sh

```diff
@@ -7,27 +7,26 @@ build_make() {
 }
 
 build_deps() {
-  if [[ "${BUILD_32BIT}" == ON ]]; then
+  if test "x${BUILD_32BIT}" = xON ; then
     DEPS_CMAKE_FLAGS="${DEPS_CMAKE_FLAGS} ${CMAKE_FLAGS_32BIT}"
   fi
-  if [[ "${FUNCTIONALTEST}" == "functionaltest-lua" ]]; then
+  if test "x${FUNCTIONALTEST}" = "xfunctionaltest-lua" ; then
```

Comment:
@jamessan `[[` is a part of the shell syntax, not a builtin. `[` and `test` are commands, in zsh this is a shell built-in (but still built-in *command*), bash and dash use `/usr/bin/[` or `/usr/bin/test`. You could not possibly emulate “looser requirements on quoting” using any command, it is a consequency of `[[` being a part of the syntax.